app/subscriber/src/features/my-reports/ReportPreview.tsx

```diff
@@ -24,6 +24,7 @@ export const ReportPreview = ({ report, onFetch, onClose }: IReportPreviewProps)
   const [, { storeReportOutput }] = useProfileStore();
   const [, { getReport, generateReport }] = useReports();
 
+  const [isLoading, setIsLoading] = React.useState(false);
```

Comment:
Needed this to stop multiple requests while a request was made to generate the preview